Automattic/jetpack
# Contact Form: add missing Required toolbar button to Checkbox field

Fixes https://github.com/Automattic/jetpack/issues/34279

## Proposed changes:

The _Checkbox_ field of the _Contact Form_ has no _Required_ toolbar button when it's set as required unlike other fields. This PR adds it.

In more details, the PR:
- extracts the existing toolbar button in its own component: `ToolbarRequiredGroup`
- replaces the existing button with this new component
- adds this new component to the _Checkbox_ field

### Other information:


## Jetpack product discussion


Project thread: pf5801-87-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Spin up a test site with this branch
- Build the block: `cd projects/packages/forms && npm run build:blocks`
- Create a new post
- Add the _Contact Form_ block
- Add a _Checkbox_ field and make sure it's selected
- Notice the _Required_ button (with an asterisk icon) in the toolbar
- Check the value is synced with the _Required_ toggle in the sidebar

<img width="939" alt="Screenshot 2023-12-13 at 3 55 45 PM" src="https://github.com/Automattic/jetpack/assets/1620183/2a187959-a8c8-438f-bdf8-e9a8effe38f4">

- Check other required fields still work as expected
